controllers/managedocs_controller.go

```diff
@@ -110,53 +142,89 @@ func (r *ManagedOCSReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error)
 	if err := r.Get(r.ctx, req.NamespacedName, r.managedOCS); err != nil {
-		return ctrl.Result{}, err
+		if errors.IsNotFound(err) {
+			r.Log.Info("No ManagedOCS resource available")
+		} else {
+			return ctrl.Result{}, err
+		}
 	}
 
 	// Run the reconcile phases
-	err := r.reconcilePhases()
+	result, err := r.reconcilePhases()
 	if err != nil {
 		r.Log.Error(err, "An error was encountered during reconcilePhases")
 	}
 
 	// Ensure status is updated once even on failed reconciles
-	statusErr := r.Status().Update(r.ctx, r.managedOCS)
+	if r.managedOCS.UID != "" {
+		statusErr = r.Status().Update(r.ctx, r.managedOCS)
+	}
 
 	// Reconcile errors have priority to status update errors
 	if err != nil {
 		return ctrl.Result{}, err
 	} else if statusErr != nil {
 		return ctrl.Result{}, statusErr
 	} else {
-		return ctrl.Result{}, nil
+		return result, nil
 	}
 }
 
-func (r *ManagedOCSReconciler) reconcilePhases() error {
-	// Update the status of the components
-	r.managedOCS.Status.Components = r.updateComponents()
+func (r *ManagedOCSReconciler) reconcilePhases() (reconcile.Result, error) {
+	if r.managedOCS.UID != "" {
+		// Update the status of the components
+		r.managedOCS.Status.Components = r.updateComponents()
 
-	// Set the effective reconcile strategy
-	reconcileStrategy := v1.ReconcileStrategyStrict
-	if strings.EqualFold(string(r.managedOCS.Spec.ReconcileStrategy), string(v1.ReconcileStrategyNone)) {
-		reconcileStrategy = v1.ReconcileStrategyNone
-	}
-	r.managedOCS.Status.ReconcileStrategy = reconcileStrategy
-
-	// Create or update an existing storage cluster
-	storageCluster := &ocsv1.StorageCluster{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      storageClusterName,
-			Namespace: r.namespace,
-		},
-	}
-	if _, err := ctrl.CreateOrUpdate(r.ctx, r, storageCluster, func() error {
-		return r.generateStorageCluster(reconcileStrategy, storageCluster)
-	}); err != nil {
-		return err
+		// Set the effective reconcile strategy
+		reconcileStrategy := v1.ReconcileStrategyStrict
+		if strings.EqualFold(string(r.managedOCS.Spec.ReconcileStrategy), string(v1.ReconcileStrategyNone)) {
+			reconcileStrategy = v1.ReconcileStrategyNone
+		}
+		r.managedOCS.Status.ReconcileStrategy = reconcileStrategy
+
+		// Create or update an existing storage cluster
+		storageCluster := &ocsv1.StorageCluster{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      storageClusterName,
+				Namespace: r.namespace,
+			},
+		}
+		if _, err := ctrl.CreateOrUpdate(r.ctx, r, storageCluster, func() error {
+			return r.generateStorageCluster(reconcileStrategy, storageCluster)
+		}); err != nil {
+			return ctrl.Result{}, err
+		}
+
+		if r.isComponentReady() &&
+			r.managedOCS.DeletionTimestamp.IsZero() &&
+			r.isUninstallValid() {
+			ok, err := r.isPvcRemoved()
+			if err != nil {
+				return ctrl.Result{}, err
+			}
+			if !ok {
+				return ctrl.Result{Requeue: true, RequeueAfter: 10 * time.Second}, nil
+			}
+			r.Log.Info("Uninstallation of OCS started")
```

Comment:
To match all other messages, we should use the same tense
```suggestion
			r.Log.Info("Starting OCS uninstallation")
```